### DISCUSSION TITLE
Beatbump Rewrite Update.

### DISCUSSION THREAD

**Discussion by snuffyDev** [snuffyDev/Beatbump#64] [Announcements]
Hello! It's been a while since I've updated the repository, and that's for good reason.

Currently the UI, and all of the underlying frontend code, have been undergoing a lot of major changes. Right now, I would not feel comfortable pushing any updates to the rewrite branch, as I do not want to risk anyone (or even myself accidentally) deploying this code live anywhere. It is very broken at the moment, but it is worth it in the long run!

The player at the bottom of the screen is getting redesigned (still VERY much a work-in-progress!)

|  *No song playing*  |
|-- |
|   <img src="https://user-images.githubusercontent.com/72365477/170016062-1d43da44-50e6-434d-82ce-040756b5dd15.jpeg" width="480"/> |

| *Song playing + song queue* |
|-- |
| <img src="https://user-images.githubusercontent.com/72365477/170016157-3e34818c-b8c9-4d50-b270-1e311150580f.jpeg" width="480"/> |
 
Along with the UI changes to the player, the actual code for it has been completely redone. It's still sort of buggy, but it's getting there!
 
Not to mention, ***Group Sessions / Listening Parties*** is heavily being worked on! So, with this feature, you will be able to listen to music with your friends -- same song queue -- and you can even activate a setting which waits for everyone to finish the song before moving on to the next! 

You can find the code for these changes [in this Gist](https://gist.github.com/snuffyDev/732f8304d76ec95a12d2ac7bda1d7430)!

I'm also working on a 'build system' so that way if anyone wants to host or deploy Beatbump, it'd be as simple as editing `beatbump.conf.json`! It'll have a schema as well, so it will be as hassle-free as possible.

![image](https://user-images.githubusercontent.com/72365477/170019184-5ec61862-f4b0-4b21-b703-7e1ba6a7cceb.png)

Of course, there are way more changes that are in the works or have otherwise been made. But keep an eye out, hopefully I should be able to commit this once a few things are ironed out!

**Comment by davidcollini** [comment]
You did great with this update! The rewrite is moving along really quickly, I can't wait to see the full release

**Comment by snuffyDev** [comment]
Cleaning up the rough edges still. But inching closer!

https://user-images.githubusercontent.com/72365477/174865911-75494299-ed92-4241-8fc9-542eebe98b9b.MOV

**Reply by davidcollini** [reply]
The new fullscreen UI gives me life, also the song queue looks so much better👌🔥🔥🔥
